apps/docs/src/guide/contracts/deploying-contracts.md

```diff
@@ -39,11 +39,11 @@ Once you have the contract artifacts, it can be passed to the `ContractFactory`
 
 As mentioned earlier, there are two different processes for contract deployment handled by the `ContractFactory`. These can be used interchangeably, however, the `deploy` method is recommended as it will automatically choose the appropriate deployment process based on the contract size.
 
-This call resolves as soon as the transaction to deploy the contract is submitted and returns three items: the `contractId`, the `transactionId` and a `waitForResult` function.
+This call resolves as soon as the transaction to deploy the contract is submitted and returns three items: the `contractId`, a `waitForTransactionId` function and a `waitForResult` function.
 
 <<< @/../../docs-snippets/src/guide/contracts/deploying-contracts.test.ts#deploy{ts:line-numbers}
 
-The `contract` instance will be returned only after calling `waitForResult` and waiting for it to resolve. To avoid blocking the rest of your code, you can attach this promise to a hook or listener that will use the contract only after it is fully deployed.
+The `contract` instance will be returned only after calling `waitForResult` and waiting for it to resolve. To avoid blocking the rest of your code, you can attach this promise to a hook or listener that will use the contract only after it is fully deployed. Similarly, the transaction ID is only available once the underlying transaction has been funded. To avoid blocking the code until the ID is ready, you can use the `waitForTransactionId` function to await it's retrieval.
```

Comment:
@danielbate, do you think it's worth mentioning the differences between normal and large contracts? 

Deploying large contracts will take more time. Each blob transaction has to wait for a block to be produced before it gets mined. After that, there's still the wait for the normal create transaction to be deployed.

Reply:
There is a note at the bottom of the doc:

<img width="722" alt="Screenshot 2024-08-14 at 16 27 28" src="https://github.com/user-attachments/assets/8b8b4f5a-70ee-458d-b0e0-b423eed59478">

But on second thoughts Im going to flesh it some more.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2929/commits/9bb81024e19c89b6dabcdff23c2b123d258ff1ef